[FE]그룹생성 페이지 - 레이아웃 잡기

ChanYangYu opened the pull request:
## 관련 이슈

- closes (#22)

## 셀프 체크리스트

> 최소 1명 이상의 assign을 받아야만 Merge 가능합니다.

- [X] Warning Message 발생 유/무
- [X] Coding Convention 준수 유/무

## 작업 내용

> 구현 및 작업 내역

- [x] 그룹생성 페이지 기본 레이아웃 구현
- [x] GageBar, InputContents 파일 생성
- [x] Contents 이전, 다음 버튼 이동 

## Preview
![image](https://user-images.githubusercontent.com/55623688/139792661-9c24b3a3-2d67-4bb4-a87c-d91a9979efa6.png)

> 선택 사항

## PR 특이 사항

> PR을 볼 때 주의깊게 봐야하거나 말하고 싶은 점

- gitignore에 .idea를 추가하는 커밋을 따로 올리기 애매해서 여기에 추가 커밋했습니다!
- 이전, 다음 버튼 로직은 InputContents 구현을 위해 임시로 구현했습니다.
- `CreateGroupPage`를 구성하는 컴포넌트는 
  - `InputContents(Category, Personnel, GroupInfo, TechStack, StartDate)` : 내부 입력 컨텐츠
  - `GageBar` :  상단 게이지바
  - `Button` : 하단 이전,다음 버튼
- 세가지로 구성되며 모두 재활용이 되지 않아서 `pages/CreateGroupPage` 폴더 내부에 모든 컴포넌트가 위치하는데 이렇게 구성하는게 맞는지 검토가 필요합니다.
```
pages/CreateGroupPage
│  GageBar.tsx
│  index.tsx
│
├─Category
│      index.tsx
│
├─GroupInfo
│      index.tsx
│
├─Personnel
│      index.tsx
│
├─StartDate
│      index.tsx
│
└─TechStack
        index.tsx

```

ChanYangYu committed:
[FE] design: 그룹생성 페이지 기본 레이아웃 구현


- 페이지 필요 Element 배치
- issue (#22)

ChanYangYu committed:
[FE] design: CreateForm 내부 Contents 레이아웃 생성


- Category, Personnel, TechStack, GroupInfo, StartDate 입력 Contents 기본 레이아웃 구성
- issue (#22)

ChanYangYu committed:
[FE] design: GageBar 레이아웃 생성


- 게이지바 컴포넌트 생성및 기본 레이아웃 구현
- issue (#22)

ChanYangYu committed:
[FE] feat: Contents 이전, 다음 이동 구현 


- 테스트를 위해 Contents 이동 로직 구현
- issue (#22)

ChanYangYu committed:
[FE] chore: gitignore .idea 추가 


`따로 올리기가 애매해서 22번 이슈에 추가했습니다!`
- .gitignore 파일에 .idea 항목 추가
- issue (#22)

ChanYangYu committed:
[FE] rename: 파일이름 및 폴더 구조 변경 


- GroupCreate -> CreateGroup
- GroupCreatePage 컴포넌트 index.tsx로 파일이름 변경
- InputComponent(카테고리, 인원수, 그룹이름...) 폴더로 묶고 index.tsx를 export하도록 구조변경
- issue (#22)

yuchanleeme commented on client/src/pages/CreateGroupPage/TechStack/index.tsx:
```diff
@@ -0,0 +1,15 @@
+const Contents = styled.div`
+  margin: 100px auto;
+  height: 400px;
+  text-align: center;
```
`text-align:center` 이 Category, Personnel, TechStack등등 과 같은 컴포넌트들에서 모두 사용되는데 상위 컨테이너에서 한번 사용해서 중복을 줄이는 방법은 어떨까요?

ChanYangYu replied:
오 생각을 못했네요ㅋㅋㅋ 바로 적용해보겠습니다.

yuchanleeme commented on client/src/pages/CreateGroupPage/index.tsx:
```diff
@@ -0,0 +1,65 @@
+  display: flex;
+  
+`
+export default CreateGroupPage;
```
지금 prettier에서 파일의 마지막 개행을 시켜주지 않는것 같습니다! EOL 관련 설정 한번 확인하셔서 파일 끝에 자동으로 개행을 넣는 옵션을 설정해주셔도 좋을것 같습니다.

ChanYangYu replied:
아하.. 마지막 개행 이런게 있군요! 적용해보겠습니다..

yuchanleeme commented on client/src/pages/CreateGroupPage/index.tsx:
```diff
@@ -0,0 +1,65 @@
+      case 2: return <TechStack/>;
+      case 3: return <GroupInfo/>;
+      case 4: return <StartDate/>;
+      default: return null;
```
switch문 다룰때 default 설정 해주시는거 너무 좋은것 같습니다. 👍

yuchanleeme commented on client/src/pages/CreateGroupPage/index.tsx:
```diff
@@ -0,0 +1,65 @@
+      <ButtonWrapper>
+        <button onClick={clickPrev}>이전</button>
+        <button onClick={clickNext}>다음</button>
+      </ButtonWrapper>
```
`clickPrev` -> `clickPrevContents`, `clickNext` -> `clickNextContents` 이렇게 (이전, 다음)이 어떤것을 가리키는지 변수명에 명시해도 좋을것 같습니다!

ChanYangYu replied:
아하 명시적으로 가리키는 것 좋습니다!

yuchanleeme reviewed (approved):
#### pages/CreateGroupPage 폴더 내부에 모든 컴포넌트가 위치하는데 이렇게 구성하는게 맞는지 검토가 필요합니다.

*  아직 어떤 내용을 넣어서 재사용이 가능할지 모르기 때문에 현재로서는 이런식으로 구성하는게 좋다고 생각합니다!
* 하지만 추후에 그룹 생성 페이지내 각 contents 별로 중복되는 구조가 생기게 된다면 그때 컴포넌트로 분리하는 방법도 있을것 같습니다.
* 이 부분은 개발 진행하면서 서로 얘기하면 좋을것 같습니다!!

구현하느라 수고많으셨습니다~~ 👍

ChanYangYu committed:
[FE] refactor: 코드 리뷰 기반 리팩토링

- Contents styled.div 컨테이너 하나로 중복코드 제거
- clickPrev, clickNext 함수 이름 구체화
  -  => clickPrevContents, clickNextContents
- EOL개행 추가

Donggggg reviewed (approved):
#### LGTM..!
+) 컴포넌트 위치 구성 고민에 대한 피드백
유찬님과 동일한 생각입니다!
개발 중에 고려를 충분히 하고 판단했으니 추후에 리팩토링이 필요하면 진행하면 좋을 것 같아요.
